gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedSource.java

```diff
@@ -92,14 +92,17 @@ protected void initLogger(SourceState state) {
     }
 
     TableType tableType = TableType.valueOf(state.getProp(ConfigurationKeys.EXTRACT_TABLE_TYPE_KEY).toUpperCase());
-    List<WorkUnitState> previousWorkunits = Lists.newArrayList(state.getPreviousWorkUnitStates());
+    SourceState previousSourceState = state.getPreviousSourceState();
+
     List<String> prevFsSnapshot = Lists.newArrayList();
 
     // Get list of files seen in the previous run
-    if (!previousWorkunits.isEmpty()
-        && previousWorkunits.get(0).getWorkunit().contains(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT)) {
-      prevFsSnapshot =
-          previousWorkunits.get(0).getWorkunit().getPropAsList(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT);
+    if (previousSourceState != null) {
+      state.setProp(ConfigurationKeys.PREVIOUS_STATE_ID_KEY,
+          previousSourceState.getProp(ConfigurationKeys.DATASET_STATE_ID_KEY, ConfigurationKeys.DEFAULT_DATASET_STATE_ID));
+      if (previousSourceState.contains(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT)) {
+        prevFsSnapshot = previousSourceState.getPropAsList(ConfigurationKeys.SOURCE_FILEBASED_FS_SNAPSHOT);
```

Comment:
I don't think this works - `DatasetState` (which is `previousSourceState`'s class) no longer supports `getProp` since #932.
Even if it does, I don't understand why this property belongs to `previousSourceState` - it seems it is added to workunits, not the dataset state.